utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/source/UniversalKafkaSourceFactory.scala

```diff
@@ -27,7 +28,7 @@ import scala.reflect.ClassTag
   * This is universal kafka source - it will handle both avro and json
   * TODO: Move it to some other module when json schema handling will be available
   */
-class UniversalKafkaSourceFactory[K: ClassTag, V: ClassTag](val schemaRegistryProvider: SchemaRegistryProvider,
+class UniversalKafkaSourceFactory[K: ClassTag, V: ClassTag](val schemaRegistryProvider: SchemaRegistryProvider[AvroSchema],
```

Comment:
this is temporary, right?

Reply:
IMO we shouldn't have `SchemaRegistryProvider[ParsedSchema]`, we should rather pass here sth like `fun[T<: ParsedSchema](schema: T): SchemaRegistryProvider[T]`. Maybe this PR can show a little bit further direction? Until now it isn't too much code.